### DISCUSSION TITLE
Unable to export forwarded port - setup by aws ssm

### DISCUSSION THREAD

**Discussion by patanjalis** [blinksh/blink#1951] [Support]
In blink.sh - “build mosh” to connect to remote machine
On the remote, setup a tunnel via ‘aws ssm start-session —target <> …”
The aws session setups a tunnel to build host port 5000. AWS session is setup using the session-manager-plugin. 
In blink.sh -  “build ports add htools 5000 50001”

When I open http://localhost:5001 on my iPad browser - it is unable to connect.

**Comment by carloscabanero** [comment]
Hi! Thanks for reaching out. I guess there is a typo with 50001 and then 5001 and it is 5001 everywhere. Ports on Build privately shared between your Build instance and your device over a Wireguard interface. They are actually quite straightforward, and it should be easy to troubleshoot:
- First, to be sure, make sure the container is actually running the service on port 5000. You can do that using `curl` or any other tool you prefer.
- Then, we need to make sure the port is exposed to the Wireguard interface inside the container. This is the most common mistake and it is probably what is going on in your case. Make sure you are looping the 5000 service on 0.0.0.0 for that container so the Wireguard interface can see it. Note that this does not expose it to the internet, it is just exposing it to the Wireguard interface outside the container.
- Reset all ports `build ports down`. Remove the ports file just to start with a clean slate (on ~/.blink-build/.ports.json). And then do again `build ports add tools 5000 5001`.
